mongo/client_encryption.go

```diff
@@ -72,6 +73,57 @@ func NewClientEncryption(keyVaultClient *Client, opts ...*options.ClientEncrypti
 	return ce, nil
 }
 
+// CreateEncryptedCollection creates a new collection with a help of automatic generation of new encryption data keys for null keyIds.
+func (ce *ClientEncryption) CreateEncryptedCollection(ctx context.Context,
+	db *Database, coll string, createOpts *options.CreateCollectionOptions,
+	kmsProvider string, dkOpts *options.DataKeyOptions) (*Collection, error) {
+	if createOpts == nil {
+		return nil, errors.New("nil CreateCollectionOptions")
+	} else if createOpts.EncryptedFields == nil {
+		return nil, errors.New("no EncryptedFields defined for the collection")
+	}
+
+	efBSON, err := transformBsoncoreDocument(db.registry, createOpts.EncryptedFields, true, "encryptedFields")
+	if err != nil {
+		return nil, err
+	}
+	r := bsonrw.NewBSONDocumentReader(efBSON)
+	dec, err := bson.NewDecoder(r)
+	if err != nil {
+		return nil, err
+	}
+	var m map[string]interface{}
+	err = dec.Decode(&m)
+	if err != nil {
+		return nil, err
+	}
+
+	if v, ok := m["fields"]; ok {
+		if fields, ok := v.(primitive.A); ok {
+			for _, field := range fields {
+				if f, ok := field.(map[string]interface{}); !ok {
+					continue
+				} else if v, ok := f["keyId"]; !ok || v == nil {
```

Comment:
```suggestion
				} else if v, ok := f["keyId"]; v == nil {
```

The condition for creating a data key requires `keyId` be present and `null`:

> Otherwise, if `F` has a ``"keyId"`` named element `K` and `K` is a ``null`` value